free-programming-books.md

```diff
@@ -306,7 +306,7 @@
 
 #### Competitive Programming
 
-* [Competitive Programming, 1st Edition](https://sites.google.com/site/stevenhalim/) (PDF)
+* [Competitive Programming, 1st Edition](http://cpbook.net/) (PDF)
```

Comment:
since only the first volume is free, I'd recommend using https://cpbook.net/#CP1details This should also fix the travis warning